### DISCUSSION TITLE
Process for updating taxonomy catalog and individual taxon files

### DISCUSSION THREAD

**Discussion by ryanmwhitephd** [NCSLI-MII/measurand-taxonomy#53] [Ideas]
Maintaining the complete catalog and the individual taxon files, unless automated, will be problematic, likely resulting in errors.
- Do we update or add the taxon files and merge into the main catalog? (Or edit the main catalog and split the files?)
- Is this somehow automated during a merge (via an action?), or made part of the release process?

**Comment by SchwartzMichael** [comment]
The Taxonomy Editor does both; merges and splits taxons from a master XML List. 
The editor has a command line argument that performs these tasks.

**Comment by ryanmwhitephd** [comment]
We first need to decide whether to make changes to individual files or the main catalog. To use the tool, we need to know whether this can be containerized and the merge (or split) process handled in an automated at task. Otherwise, we depend on this manual step.
Probably should have a simple script rather than entire editor. @dwloup69 was going to look into this. A possible solution is here: https://github.com/peter-evans/create-pull-request.
A possible workflow:
- Commit changes to individual taxon files.
- Create pull/merge request
- Review/approve changes
- Run action to merge files and commit to main catalog
- Approve merge into main branch

**Comment by ryanmwhitephd** [comment]
Suggested workflow in previous comment.

**Comment by SchwartzMichael** [comment]
We need to edit the individual files. That way, the comments on the check-in are localized to the changes in that file. 

I look at the Editor as a compiler and all the individual XML files as the source code.  
Once we are ready for a release, all the source XML files will be compiled into one master file.  Versioned as release x.x

**Comment by ryanmwhitephd** [comment]
Ok, so part of the release process. That we can manage now. 
Updates and new taxons:
- Commit changes to individual source files
- Create pull/merge request
- Approve request
- Merge into main

For release
- Create a release issue
- Create branch for release
- Merge individual source files into main catalog
- Commit main catalog
- Create pull/merge request (only change is the updated main catalog)
- Approve merge
- Merge updated catalog to main branch
- Once merged, create tag and make release

**Comment by MarkKuster** [comment]
This makes sense to me. Colin should probably look at this also.

**Comment by colinindy** [comment]
Yes. This looks good.

**Comment by ryanmwhitephd** [comment]
Implemented. Process, scripts and actions should be documented though.
